### ISSUE TITLE
Move all providers to crossplane-contrib

### ISSUE THREAD

**Issue by negz** [crossplane/crossplane#3054]
### What problem are you facing?

Under the recently merged [project charter](https://github.com/crossplane/crossplane/blob/6c1b06507db47801c7a1c7d91704783e8d13856f/CHARTER.md) extensions such as providers are no longer 'in scope' for Crossplane as a project. Per https://github.com/crossplane/crossplane/pull/3053 we'd like to keep https://github.com/crossplane scoped to things that fall under the charter, while https://github.com/crossplane-contrib will serve as a neutral home (under the project's governance) for things that are 'out of scope' like providers. This is similar to https://github.com/kubernetes and https://github.com/kubernetes-sigs.

Some of the more popular providers are currently under the https://github.com/crossplane organization, whereas under the new charter they would fit better under https://github.com/crossplane-contrib.

### How could Crossplane help solve your problem?

We could move the following providers to crossplane-contrib:

* [x] https://github.com/crossplane/provider-aws
* [x] https://github.com/crossplane/provider-azure
* [x] https://github.com/crossplane/provider-gcp
* [x] https://github.com/crossplane/provider-gcp-beta
* [x] https://github.com/crossplane/provider-alibaba
* [x] https://github.com/crossplane/provider-rook

GitHub will take care of redirecting the web URLs and any existing git repos so I'm _fairly_ sure this should not be a disruptive change.

**Comment by MisterMX** [comment]
Will the docker images be moved as well?

Currently, all provider images, regardless if in `crossplane` or `crossplane-contrib` have the tag `crossplane/...`

**Comment by negz** [comment]
@MisterMX They'll stay the same (`crossplane/provider-*`) for the foreseeable future, I believe.

**Comment by negz** [comment]
Note to my future self (or whoever takes this) - we'll need to make sure the repos still have access to all the GH secrets they need to build and publish once they're under the new org. I'm pretty sure all those secrets exist at the crossplane-contrib GH org level and we'll just need to give the new repos permission to consume them.

**Comment by muvaf** [comment]
> They'll stay the same (`crossplane/provider-*`) for the foreseeable future, I believe.

@negz I'd actually propose that we have a `crossplane-contrib` org in DockerHub for these images. It'd be helping with differentiation laid out in the charter and also be easier to discover the source repo from the image and vice versa.

**Comment by negz** [comment]
I'm going to start this now.

**Comment by negz** [comment]
This process is proving to be dramatically more painful than I expected, due to the need to rename the Go modules and update all the imports. Currently working on provider-aws, which is blocked on https://github.com/aws-controllers-k8s/code-generator/pull/338.
